apps/src/templates/SchoolInfoInputs.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { Component, PropTypes } from 'react';
+import CountryAutocompleteDropdown from '@cdo/apps/templates/CountryAutocompleteDropdown';
+import SchoolTypeDropdown from '@cdo/apps/templates/SchoolTypeDropdown';
+import SchoolAutocompleteDropdownWithLabel from '@cdo/apps/templates/census2017/SchoolAutocompleteDropdownWithLabel';
+import SchoolNotFound from '@cdo/apps/templates/SchoolNotFound';
+import i18n from "@cdo/locale";
+
+const SCHOOL_TYPES_HAVING_NCES_SEARCH = ['charter', 'private', 'public'];
+
+const SCHOOL_TYPES_HAVING_NAMES = [
+  'charter',
+  'private',
+  'public',
+  'afterschool',
+  'organization',
+];
+
+export default class SchoolInfoInputs extends Component {
+  static propTypes = {
+    onCountryChange: PropTypes.func.isRequired,
+    onSchoolTypeChange: PropTypes.func.isRequired,
+    onSchoolChange: PropTypes.func.isRequired,
+    onSchoolNotFoundChange: PropTypes.func.isRequired,
+    country: PropTypes.string,
+    schoolType: PropTypes.string,
+    ncesSchoolId: PropTypes.string,
+    schoolName: PropTypes.string,
+    schoolCity: PropTypes.string,
+    schoolState: PropTypes.string,
+    schoolZip: PropTypes.string,
+    useGoogleLocationSearch: PropTypes.bool,
+    fieldNames: PropTypes.object,
+  };
+
+  static defaultProps = {
+    schoolType: "",
+    country: "",
+    ncesSchoolId: "",
+    schoolName: "",
+    schoolCity: "",
+    schoolState: "",
+    schoolZip: "",
+    useGoogleLocationSearch: false,
+    fieldNames: {
+      schoolType: "user[school_info_attributes][school_type]",
+      country: "user[school_info_attributes][country]",
+      ncesSchoolId: "user[school_info_attributes][school_id]",
+      schoolName: "user[school_info_attributes][school_name]",
+      schoolState: "user[school_info_attributes][school_state]",
+      schoolZip: "user[school_info_attributes][school_zip]",
+      googleLocation: "user[school_info_attributes][full_address]",
+    },
+  };
+
+  render() {
+    const isUS = this.props.country === 'United States';
+    const outsideUS = !isUS;
+    const ncesInfoNotFound = (this.props.ncesSchoolId === '-1');
+    const noDropdownForSchoolType = (
+      !SCHOOL_TYPES_HAVING_NCES_SEARCH.includes(this.props.schoolType)
+      && this.props.schoolType !== ''
+    );
+    const askForName = SCHOOL_TYPES_HAVING_NAMES.includes(this.props.schoolType);
+    const schoolNameLabel = ['afterschool', 'organization'].includes(this.props.schoolType)
+                          ? i18n.signupFormSchoolOrOrganization()
+                          : i18n.schoolName();
+
+    return (
+      <div style={{width: 600}}>
```

Comment:
This width is wider than some mobile screens; I don't think we want to hardcode it for uses of this component on responsive pages.

Reply:
Yeah, I don't love this either. It is effectively what the sign up form does (using a style with a width of 600.)  CountryAutocompleteDropdown also has [a fixed width](https://github.com/code-dot-org/code-dot-org/blob/14fdb203cf7f3284b36e8f0b400b574ca052a13f/apps/src/templates/CountryAutocompleteDropdown.jsx#L12) for part of its layout. Getting it to display well for arbitrary widths was more than I was ready to tackle right now. 

How do you feel about leaving this set to 600 for now? It is one of the things that will need to be updated as this component gets reused in different places.